packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -644,6 +645,179 @@ export function authenticateWithPowerPlatformConnectorPlugins(
+
+  return callFunctionInHost(
+    ApiName.ExternalAppAuthentication_AuthenticateWithConnector,
+    [new SerializableConnectorParameters(input)],
+    getApiVersionTag(
+      externalAppAuthenticationTelemetryVersionNumber,
+      ApiName.ExternalAppAuthentication_AuthenticateWithConnector,
+    ),
+    isInvokeError,
+  );
+}
+
+/**
+ * @hidden
+ * Represents the possible authentication states for a user in the context of federated connectors.
+ *
+ * @enum {number}
+ * @property {number} Invalid - The authentication state is invalid or not established.
+ * @property {number} Valid - The authentication state is valid and the user is authenticated.
+ * @property {number} Expired - The authentication state has expired and re-authentication is required.
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export enum UserAuthenticationState {
+  /** The authentication state is invalid or not established. */
+  Invalid,
+  /** The authentication state is valid and the user is authenticated. */
+  Valid,
+  /** The authentication state has expired and re-authentication is required. */
+  Expired,
+}
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * Response handler for user authentication state for federated connectors.
+ * Validates and deserializes the response from the host to ensure it matches the UserAuthenticationState enum.
+ */
+class UserAuthenticationStateResponseHandler extends ResponseHandler<UserAuthenticationState, UserAuthenticationState> {
+  /**
+   * Validates if the response is a valid UserAuthenticationState value.
+   * @param response - The response received from the host.
+   * @returns True if the response is a valid UserAuthenticationState, false otherwise.
+   */
+  public validate(response: unknown): response is UserAuthenticationState {
+    return typeof response === 'string' && response in UserAuthenticationState;
+  }
+
+  /**
+   * Deserializes the response to a UserAuthenticationState value.
+   * @param response - The response received from the host.
+   * @returns The response cast as UserAuthenticationState.
+   */
+  public deserialize(response: unknown): UserAuthenticationState {
+    return response as UserAuthenticationState;
+  }
+}
+
+/**
+ * @beta
+ * @hidden
+ * Retrieves the authentication state for a user for a given federated connector.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @param connectorId - The unique identifier for the federated connector.
+ * @param oauthConfigId - The OAuth configuration ID associated with the connector.
+ * @returns A promise that resolves to the user's authentication state for the federated connector.
+ * @throws Error if the capability is not supported or if initialization has not completed.
+ */
+export function getUserAuthenticationStateForConnector(input: ConnectorParameters): Promise<UserAuthenticationState> {
+  ensureInitialized(runtime, FrameContexts.content);
```

Comment:
Are we restricting this to content frame context?

Reply:
yes.